rasa/dialogue_understanding/processor/command_processor.py

```diff
@@ -113,6 +148,16 @@ def execute_commands(
 
     commands = clean_up_commands(commands, tracker, all_flows)
 
+    updated_flows = find_updated_flows(tracker, all_flows)
+    if updated_flows:
+        # Override commands
+        commands = [CleanStackCommand()]
+
+    # store current flow hashes if they changed
+    new_hashes = calculate_flow_fingerprints(all_flows)
+    if new_hashes != (tracker.get_slot(FLOW_HASHES_SLOT) or {}):
+        tracker.update_with_events([SlotSet(FLOW_HASHES_SLOT, new_hashes)], None)
```

Comment:
mhm should we rather add this to `events`? I don't like the implicitness of updating the tracker here and hoping that the event will be persisted

Reply:
Ah okay, I was confused bc for the actions you apply events here but then you also apply them again at the calling method. But I guess it's okay since they are idempotent?